dashboard/common/db_util.py

```diff
@@ -110,9 +108,32 @@ def get_user_courses_info(username: str) -> List[Dict[str, Union[str, int, List[
                 course_enrollments[course_id]['enrollment_types'].append(enrollment_type)
             user_courses_info = list(course_enrollments.values())
     logger.info(f'User {username} is enrolled in these courses: {user_courses_info}')
+
+    course_id_to_filter = get_course_id_from_request_url(request)
```

Comment:
A few questions about this method:
It doesn't look like you need to pass the entire request into this method, only the request.path.

This also doesn't make sense logically anymore.
1) It queries for all of the users courses from the database
2) You find which course the user is in
3) You remove everything but that course

It seems like this isn't "get_user_courses_info" anymore but it's get_course_info or something more specific. 

Are we changing this so the user can't switch courses anymore in MyLA? What about if it's not launched with LTI? 

And really switching courses isn't technically a problem since we're just using MyLA for auth, it just might be confusing since it's in the context of a course in Canvas.

I think this should work both ways and it for this case is should just do a single database call to get this specific course rather than filtering it afterward. 

Unless I'm mistaken on the purpose of these new methods.

Reply:
> It doesn't look like you need to pass the entire request into this method, only the request.path.

Done

Reply:
> Are we changing this so the user can't switch courses anymore in MyLA? What about if it's not launched with LTI?

If course members launches myla from canvas they will only see the single course info as per launch. Myla might carry more enrollment data if user is in multiple courses in MYLA DB. So the filtering is based on the /course/12333 and STUDENT_DASHBOARD_LTI enabled options

When the tool is standalone they will see their default courses dashboard they belong to. This is unchanged

Reply:
> It seems like this isn't "get_user_courses_info" anymore but it's get_course_info or something more specific.

The DB query is still based on username. I still feel the name is still relevant

Reply:
> And really switching courses isn't technically a problem since we're just using MyLA for auth, it just might be confusing since it's in the context of a course in Canvas.

I am not sure if I understood you comment, Admin can still switch courses from MYLA even in LTI (in standalone the feature still the same) and admin's can become user

Reply:
> I think this should work both ways and it for this case is should just do a single database call to get this specific course rather than filtering it afterward.

@jonespm I am now not pulling all course info from DB and later filtering based on the need (LTI vs standalone. I think I resolved your concern
[`16635fb` (#1010)](https://github.com/tl-its-umich-edu/my-learning-analytics/pull/1010/commits/16635fbe0daebe4075e517d8c315f99ffc7c246f)

Reply:
Okay great.